### PR TITLE
Feature/user profile

### PR DESCRIPTION
**Pull request by ManuViola77**
This PR is to implement the User Story [Edit Profile](https://trello.com/c/rI8YkwIB/8-3-as-a-user-i-should-be-able-to-edit-my-profile-so-i-can-update-my-personal-information-or-fill-incomplete-fields).

For now, it just updates username, email and first and last name. In the next step it will be able to update password and avatar (photo) too.

### CODE REVIEW

File: src/validations/updateProfileValidations.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {
+  name,
+  email,
+  firstName,
+  lastName,
+  gender,
+  avatar,
+  //push_token,
```
Comment (germanStabile) [inline src/validations/updateProfileValidations.js]: remove commented code or add TODO

File: src/validations/updateProfileValidations.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+} from 'constants/fields';
+
+import strings from 'locale';
+
+const { SIGN_UP_ERROR /*, UPDATE_PROFILE_ERROR*/ } = strings;
```
Comment (germanStabile) [inline src/validations/updateProfileValidations.js]: same here

File: src/navigators/index.js [hunk 3]
```diff
@@ -28,11 +31,33 @@ const AuthNavigator = createStackNavigator(
...
+        borderBottomWidth: 0,
+      },
+    }),
+  },
+  ProfileScreen: {
+    screen: ProfileScreen,
+    navigationOptions: ({ navigation }) => ({
+      headerLeft: null,
+      headerRight: (
+        <TouchableOpacity onPress={() => navigation.goBack()}>
```
Comment (peter7z) [inline src/navigators/index.js]:
```suggestion
        <TouchableOpacity onPress={navigation.goBack}>
```
Only if this isn't passing parameters that cause unwanted side effects...

File: src/reducers/profileReducer.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { createReducer } from '@rootstrap/redux-tools';
+import { getProfileSuccess } from 'actions/profileActions';
+
+const initialState = { user: {} };
+
+const profileReducer = {
+  [getProfileSuccess]: (store, action) => {
+    store.user = action.payload;
```
Comment (peter7z) [inline src/reducers/profileReducer.js]:
```suggestion
  [getProfileSuccess]: (store, { payload }) => {
    store.user = payload;
```

File: src/screens/ProfileScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+      />
+      <Input
+        callback={newValue => handleChange(lastNameField, newValue)}
+        errorMessage={errors[lastNameField]}
+        text={values[lastNameField]}
+        title={PROFILE.lastName}
+      />
+      <Link
+        text={PROFILE.password}
+        onPress={() => console.log('change password')}
```
Comment (peter7z) [inline src/screens/ProfileScreen/index.js]: 😄

File: src/navigators/index.js (hunk 3 above)
Comment (ManuViola77) [reply src/navigators/index.js]: yes, is that case so I leave it like this. But good to know for other cases.

File: src/reducers/profileReducer.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  },
+};
+
+export default createReducer(initialState, profileReducer);
```
Comment (lucia-santos) [inline src/reducers/profileReducer.js]: Do you find it necessary to create this profile reducer? Couldn't you store this information in the `sessionReducer`?

**Pull request merged**